charts/opentelemetry-collector/templates/networkpolicy.yaml

```diff
@@ -0,0 +1,39 @@
+{{- if .Values.networkPolicy.enabled }}
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: {{ include "opentelemetry-collector.fullname" . }}
+  labels:
+    {{- include "opentelemetry-collector.labels" . | nindent 4 }}
+  {{- if .Values.annotations }}
+  annotations:
+    {{- range $key, $value := .Values.annotations }}
+    {{- printf "%s: %s" $key (tpl $value $ | quote) | nindent 4 }}
+    {{- end }}
+  {{- end }}
+spec:
+  podSelector:
+    matchLabels:
+      {{- include "opentelemetry-collector.selectorLabels" . | nindent 6 }}
+      {{- include "opentelemetry-collector.component" . | nindent 6 }}
+  ingress:
+    - ports:
+        {{- range $key, $port := .Values.ports }}
+        {{- if $port.enabled }}
+        # {{ $key }}
```

Comment:
Can we remove this comment if not needed?

Reply:
Done, I also removed the `$key` as that's just extra if the comment is gone.